sync-core/src/main/java/com/cloudant/sync/datastore/BasicDatastore.java

```diff
@@ -1703,7 +1780,7 @@ private static BasicDocumentRevision getFullRevisionFromCurrentCursor(Cursor cur
             parent = cursor.getLong(7);
         } else if(cursor.columnType(7) == Cursor.FIELD_TYPE_NULL) {
         } else {
-            throw new IllegalArgumentException("Unexpected type: " + cursor.columnType(7));
+            throw new IllegalArgumentException("Unexpected type: " + cursor.columnType(7)); //TODO Should be straight Runtime Exception? Definitely not an argument
```

Comment:
I assume at some point `cursor` may have been provided from a query that didn't include `parent` in the list of columns to get.

As far as I can see, this is never the case any more (though it's really quite broken that this method has to know the order of columns in the cursor and special case one).

I suggest assuming `parent` is present and deleting the `if` and seeing if the tests fail.